packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -113,114 +104,216 @@ describe("Presence", () => {
+								const attendee = presence.getAttendee(id);
+
+								// Verify
+								assert.equal(attendee, knownAttendee, "getAttendee returned wrong attendee");
+								assert.equal(
+									attendee.getConnectionStatus(),
+									SessionClientStatus.Disconnected,
+									"getAttendee returned attendee with wrong status",
+								);
+							});
+						});
+					}
+
+					describe("is available from `getAttendees`", () => {
+						it('with status "Connected"', () => {
+							// Setup
+							assert(knownAttendee !== undefined, "No attendee was set in beforeEach");
+
 							// Act
-							const attendee = presence.getAttendee(id);
+							const attendees = presence.getAttendees();
+							assert(
+								attendees.has(knownAttendee),
+								"getAttendees set does not contain attendee",
+							);
+							assert.equal(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Connected,
+								"getAttendees set contains attendee with wrong status",
+							);
+						});
+
+						it('with status "Disconnected"', () => {
+							// Setup
+							assert(knownAttendee !== undefined, "No attendee was set in beforeEach");
+
+							// Act - remove client connection id
+							presence.removeClientConnectionId(initialAttendeeConnectionId);
+
 							// Verify
-							assert.equal(attendee, newAttendee, "getAttendee returned wrong attendee");
+							const attendees = presence.getAttendees();
+							assert(
+								attendees.has(knownAttendee),
+								"getAttendees set does not contain attendee",
+							);
+							assert.equal(
+								knownAttendee.getConnectionStatus(),
+								SessionClientStatus.Disconnected,
+								"getAttendees set contains attendee with wrong status",
+							);
 						});
-					}
+					});
+
+					it('is not announced via `attendeeJoined` when already "Connected"', () => {
+						// Setup
+						afterCleanUp.push(
+							presence.events.on("attendeeJoined", () => {
+								assert.fail("No attendee should be announced in beforeEach");
+							}),
+						);
 
-					it("is available from `getAttendees`", () => {
+						// Act - simulate join message from client
+						presence.processSignal("", initialAttendeeSignal, false);
+					});
+
+					describe("has their connection removed", () => {
 						// Setup
-						assert(newAttendee !== undefined, "No attendee was set in beforeEach");
+						let disconnectedAttendee: ISessionClient | undefined;
+						beforeEach(() => {
+							afterCleanUp.push(
+								presence.events.on("attendeeDisconnected", (attendee) => {
+									assert(
+										disconnectedAttendee === undefined,
+										"Only one attendee should be disconnected",
+									);
+									disconnectedAttendee = attendee;
+								}),
+							);
+
+							// Act - remove client connection id
+							presence.removeClientConnectionId(initialAttendeeConnectionId);
+
+							it("is announced via `attendeeDisconnected`", () => {
```

Comment:
Something seems off - there is an action taken in same scope as new test case?

Reply:
Updated, accidently had the test cases in the before each block